subt/artf_filter.py

```diff
@@ -31,12 +33,22 @@ def maybe_remember_artifact(self, artifact_data, artifact_xyz):
                         print('False detection - dist:', distance3D((x, y, z), artifact_xyz))
                     return False
 
-        for stored_data, (x, y, z) in self.artifacts:
+        for i, (stored_data, (x, y, z)) in enumerate(self.artifacts):
             if distance3D((x, y, z), artifact_xyz) < 4.0:
                 # in case of uncertain type, rather report both
                 if stored_data == artifact_data:
-                    return False
+                    self.confirmations[i] += 1
+                    # return true only when confirmation threshold was reached
+                    if self.verbose:
+                        print('Confirmed:', artifact_data, self.confirmations[i])
+                    if artifact_data == GAS:
+                        return False  # ignore confirmation - virtual detector is perfect
+                    return self.confirmations[i] == self.num_confirm
         self.artifacts.append((artifact_data, artifact_xyz))
+        self.confirmations.append(0)
```

Comment:
So 1 confirmation = 2 observations? This is somewhat confusing.

Reply:
ACK - so maybe instead of `confirmations` I should use `observations`